sklearn/ensemble/_hist_gradient_boosting/histogram.pyx

```diff
@@ -177,6 +177,12 @@ cdef class HistogramBuilder:
                 self.ordered_hessians[:n_samples]
             unsigned char hessians_are_constant = \
                 self.hessians_are_constant
+            unsigned int bin_idx = 0
+        
```

Comment:
Maybe add a comment to explain that we do a delayed init here to leverage feature-wise thread-based parallelism in the Cython code.

Reply:
Ok, added that earlier next to np.empty